Secret Outputs and Env Vars Masking Fix

LanceMcCarthy opened the pull request:
This PR contains the following modification

- I added a workflow to run and validate each commit to main release branch
- Updated secrets.js with the masking technique before setting output and env var exports. 
- Reverted the tests to check for outputs and env vars

I have 100% confirmed this is working in my test workflows, here's the output:

![image](https://user-images.githubusercontent.com/3520532/187778560-dfde6bba-b7c5-4bed-80b4-c325d9fe71e7.png)

You can see the results for yourself here https://github.com/LanceMcCarthy/akeyless-action/actions/runs/2966939880

LanceMcCarthy committed:
Added node_modules to gitignore

We should never need to check-in the dependencies folder with the source code. This will get restored when npm install is run on every end system.

LanceMcCarthy committed:
Update secrets.js

This contains the potential fix for https://github.com/cmancone/akeyless-action/issues/2

LanceMcCarthy committed:
Update tests, cleanup package-lock

LanceMcCarthy committed:
Create ci.yml

Add a simple workflow to ensure tests pass when main or a v* release branch is created

LanceMcCarthy committed:
Merge pull request #1 from LanceMcCarthy/lance/fix-secret-masking

Merge secret masking updates and other improvements

LanceMcCarthy committed:
Update ci.yml

LanceMcCarthy committed:
Update ci.yml

LanceMcCarthy committed:
Add node_modules back to repo for distribution consideration

LanceMcCarthy committed:
Update ci.yml

LanceMcCarthy commented:
[edit] This is outdated, I have made the changes for you in the PR


~~After you merge this PR, you will probably want to update the **ci.yaml** workflow file with your own values for:~~

1. Change `access-id` to yours (or add a GitHub secret like I use, it's on the Settings > Secrets > Actions page)
2. Change `/personal-keys/mccarthy/MyFirstSecret` to `/MyFirstSecret`

```
    - name: Fetch static secrets from AKeyless
      id: fetch-static-secrets
      uses: ./   # IMPORTANT - this references './' instead of the Marketplace's listing
      with:
        access-id: ${{ secrets.AKEYLESS_ACCESS_ID }}
        static-secrets: '{"/personal-keys/mccarthy/MyFirstSecret":"my_first_secret"}'
        export-secrets-to-outputs: true
        export-secrets-to-environment: true
```

cmancone commented on secrets.test.js:
```diff
@@ -26,17 +25,16 @@ test('export dynamic secrets', async () => {
         'token': 'akeyless-token',
         'name': '/path/to/dynamic/producer',
     });
-    expect(core.setSecret).toHaveBeenCalledWith('sup', dynamicSecret);
```
You are calling setSecret - that's the part that makes it a secret, which is also the important part, so you may as well leave this in and test that both `setSecret` and `setOuptut` are called.

LanceMcCarthy replied:
[edit] I originally had all three setSecret, setOutput and exportVariable in the update.  I'm pushing changes that resolve the remaining items now.

cmancone commented on secrets.test.js:
```diff
@@ -13,9 +13,8 @@ test('export dynamic secrets', async () => {
-
+    core.setOutput = jest.fn(() => {});
     core.exportVariable = jest.fn(() => {});
-    core.setSecret = jest.fn(() => {});
```
Related to the comment below (I went out of order): since `core.setOutput` is being called and `core.setSecret` is being called, and they are both important, let's just test that both are called.

cmancone commented on secrets.test.js:
```diff
@@ -47,6 +45,6 @@ test('export dynamic secrets', async () => {
         'token': 'akeyless-token',
         'names': ['/path/to/static/secret'],
     });
-    expect(core.setSecret).toHaveBeenCalledWith('sup', 'super secret');
```
Same with the above - let's test that both setSecret and setOutput are both being properly called.

cmancone commented on secrets.js:
```diff
@@ -31,13 +35,18 @@ function getStaticSecret(api, name, variableName, akeylessToken, exportSecretsTo
             'names': [name],
         })).then(staticSecret => {
+            // Mask secret value in output
+            var secretValue = staticSecret[name];
```
var -> const, e.g.:

```
const secretValue = staticSecret[name];
```

Ref: https://codeburst.io/javascript-var-let-or-const-which-one-should-you-use-2fd521b050fa

LanceMcCarthy replied:
Agreed, constant is better here (var was just memory muscle).

cmancone commented:
Yeah, sorry, I didn't have a chance to review this yesterday.  The changes I requested are pretty minor though - if you can make those, I'll make sure and merge it in ASAP.

LanceMcCarthy commented:
No worries! I'll be online in about 15m and will push the changes. Thank you for being responsive.

LanceMcCarthy committed:
Update secrets.js

LanceMcCarthy committed:
Update secrets.test.js

- Updated tests to include all three used Actions core methods
- Update name of static test method (accidentally was a clone of the dynamic test).

LanceMcCarthy committed:
Update ci.yml

LanceMcCarthy commented:
@cmancone All set, please let me know if there's anything else I can do.

For you convenience, I also updated the ci.yaml action to use your demo's values. The only remaining thing for you would be to bump the version number in action.yml and publish a new release.

LanceMcCarthy commented:
I just saw the ci workflow has failed, this is probably because of the access-id. Does it still exist? 

![image](https://user-images.githubusercontent.com/3520532/187932063-105d7fb0-f846-4993-bc0a-66ce302a3f42.png)


I noticed that the other repository you use for the demo hasn't been run in a few months, so maybe the access-id is outdated?
